cylc/flow/task_job_mgr.py

```diff
@@ -767,7 +767,10 @@ def _manip_task_jobs_callback(
         tasks = {}
         for itask in itasks:
             while itask.reload_successor is not None:
```

Comment:
At this point:
- `suite.rc`: `itask.reload_successor` is a `TaskProxy` object
- `flow.cylc`: `itask.reload_successor` is `None`

Reply:
I'll look at this more closely tomorrow, post webinar. But not that back-compat mode does have different task pool content: the downstream task in the example gets pre-spawned as waiting.

Reply:
(This is all fine, a consequence of the aforementioned spawning difference).